stripe/src/main/java/com/stripe/android/cards/CardNumber.kt

```diff
@@ -0,0 +1,74 @@
+package com.stripe.android.cards
+
+import com.stripe.android.StripeTextUtils
+
+internal sealed class CardNumber(number: String) {
+    private val normalizedNumber = StripeTextUtils.removeSpacesAndHyphens(number).orEmpty()
+
+    /**
+     * A representation of a partial or full card number that hasn't been validated.
+     */
+    class Unvalidated(number: String) : CardNumber(number)
+
+    /**
+     * Format a number based on its expected length
+     *
+     * e.g. `"4242424242424242"` with pan length `16` will return `"4242 4242 4242 4242"`;
+     * `"424242"` with pan length `16` will return `"4242 42"`;
+     * `"4242424242424242"` with pan length `14` will return `"4242 424242 4242"`
+     */
+    fun getFormatted(
+        panLength: Int = DEFAULT_PAN_LENGTH
+    ) = formatNumber(panLength, getSpacePositions(panLength))
```

Comment:
Is the plan for `formatNumber` to be used by additional functions later? Is there another reason you're delegating here? If not, I would get rid of the delegation and just call `getSpacePositions(panLength)` inside the function.